liquidhaskell-boot/src/Language/Haskell/Liquid/Types/Names.hs

```diff
@@ -231,7 +254,19 @@ getLHNameSymbol (LHNUnresolved _ s) = s
 -- | Get the unresolved Symbol from an LHName.
 getLHNameResolved :: HasCallStack => LHName -> LHResolvedName
 getLHNameResolved (LHNResolved n _) = n
-getLHNameResolved n@LHNUnresolved{} = error $ "getLHNameResolved: unresolved name: " ++ show n
+getLHNameResolved n@LHNUnresolved{} = error $ "getLHNameResolved: unresolved name: " ++ showLHNameDebug n
+
+showLHNameDebug :: LHName -> String
```

Comment:
This should be the `Show` instance really, and what is now the instance should be a regular function that explains what it is for.

Reply:
I tried that, and found that it influenced error messages intended for users. I'll address this in a follow up PR.